core/src/main/java/bio/terra/pearl/core/service/survey/SurveyResponseService.java

```diff
@@ -193,50 +193,54 @@ public HubResponse<SurveyResponse> updateResponse(SurveyResponse responseDto, Re
                                                       Enrollee enrollee, UUID taskId, UUID portalId) {
 
 
-        ParticipantTask task = participantTaskService.authTaskToEnrolleeId(taskId, enrollee.getId()).orElseThrow(() -> new NotFoundException("Task not found or not authorized for enrollee %s and task %s".formatted(enrollee.getId(), taskId)));
+        ParticipantTask task = participantTaskService.authTaskToEnrolleeId(taskId, enrollee.getId(), true).orElseThrow(() -> new NotFoundException("Task not found or not authorized for enrollee %s and task %s".formatted(enrollee.getId(), taskId)));
 
         Survey survey = surveyService.findByStableIdWithMappings(task.getTargetStableId(),
                 task.getTargetAssignedVersion(), portalId).get();
         validateResponse(survey, task, responseDto.getAnswers());
 
 
-        SurveyResponse priorResponse = dao.findOneWithAnswers(task.getSurveyResponseId()).orElse(null);
-        SurveyResponse response;
+        SurveyResponse response = task.getSurveyResponseId() != null ?
+                dao.findOneWithAnswers(task.getSurveyResponseId()).orElseThrow() : null;
 
         if (survey.getRecurrenceType() == RecurrenceType.LONGITUDINAL
-                && priorResponse != null
-                && !isLongitudinalTaskEditable(enrollee, priorResponse, survey.getStableId())
-                // admins should be able to update old responses
-                && operator.getParticipantUser() != null) {
-            throw new IllegalArgumentException("Cannot update previous responses for longitudinal surveys");
+         && operator.getParticipantUser() != null && response != null) {
+            // handle participant response to a longitudinal survey with prior response
+            if (!isLongitudinalTaskEditable(enrollee, response, survey.getStableId())) {
+                // only admins can edit old longitudinal responses
```

Comment:
comment no longer applicable here; more accurately would be at the top if statement: `admins do not trigger longitudinal logic`